roles/nat64_appliance/tasks/main.yml

```diff
@@ -0,0 +1,54 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure nat64-appliance working directory
+  ansible.builtin.file:
+    path: "{{ cifmw_nat64_appliance_dir }}"
+    state: directory
+    mode: '0755'
+
+- name: Make sure python3-pip is installed
+  become: true
+  ansible.builtin.package:
+    name:
+      - python3-pip
+      - qemu-img
+    state: present
+
+- name: Install diskimage-builder in virtualenv
+  ansible.builtin.pip:
+    virtualenv_command: "python3 -m venv"
+    virtualenv: "{{ cifmw_nat64_appliance_dir }}/venv"
+    name:
+      - diskimage-builder
+      - setuptools
+
+- name: Copy files to cifmw_nat64_appliance_dir
+  ansible.builtin.copy:
+    src: "{{ item }}"
+    dest: "{{ cifmw_nat64_appliance_dir }}/{{ item }}"
+  loop:
+    - elements/
+    - nat64-appliance.yaml
+
+- name: Build the nat64-appliance image using DIB
+  environment:
+    ELEMENTS_PATH: "{{ cifmw_nat64_appliance_dir }}/elements"
+    DIB_IMAGE_CACHE: "{{ cifmw_nat64_appliance_dir }}/cache"
+  ansible.builtin.command:
```

Comment:
I think "nice" logs would be good for that run. Would it be possible to use `cifmw.general.ci_script` instead?
Here's an example:
https://github.com/openstack-k8s-operators/ci-framework/blob/9869c491e6cff0512cc4f5bcad2246393b737947/roles/edpm_prepare/tasks/kustomize_and_deploy.yml#L50-L56

You must ensure the following directories exist before:
- `"{{ cifmw_nat64_appliance_basedir }}/artifacts"`
- `"{{ cifmw_nat64_appliance_basedir }}/logs"`

In this case, the `_basedir` would be useful as well :). The ci_script will create a reproducer script in artifacts, and push the log file in logs.

Reply:
Done